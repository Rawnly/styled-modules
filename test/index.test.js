import _transform from './_transform';
import plugin from '../src/babel';

const transform = (file, opts) => _transform(file, {
  plugins: [
    [plugin, { pattern: /\.css$/ }]
  ],
  ...opts,
});

it('works', async () => {
  const { code } = await transform('./fixtures/index.fixture.js');
  expect(code).toMatchSnapshot();
});
