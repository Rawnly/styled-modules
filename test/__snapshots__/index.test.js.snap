// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not do anything 1`] = `
"require('babel-polyfill');
import React from 'react';
const Header = require('./Header');

export default class extends React.Component {
  render() {
    return <Header />;
  }
}"
`;

exports[`should not inject styled-modules/style 1`] = `
"import React from 'react';
import style from 'styled-modules/style';
const StyledModule = require('styled-modules/style');
import layoutStyles from './Layout.css';
import contentStyles from './Content.css';

const Content = class extends React.Component {
  render() {
    return <_StyledModules styles={[{
      __hash: layoutStyles.__hash,
      __css: layoutStyles.__css
    }, {
      __hash: contentStyles.__hash,
      __css: contentStyles.__css
    }]}>
      <style styles={[contentStyles]}>
        <div className={contentStyles.content}>
          Content
        </div>
      </style>
    </_StyledModules>;
  }
};

export default (() => <_StyledModules styles={[{
  __hash: layoutStyles.__hash,
  __css: layoutStyles.__css
}, {
  __hash: contentStyles.__hash,
  __css: contentStyles.__css
}]}>
  <StyledModule styles={[layoutStyles]}>
    <div className={layoutStyles.layout}>
      <Content />
    </div>
  </StyledModule>
</_StyledModules>);"
`;

exports[`works 1`] = `
"import _StyledModules from 'styled-modules/style';
import React from 'react';
import _globalStyles from './global.css';

var _globalStyles2 = require('./global2.css');

import styles from './helloWorld.css';
import _styles, { root } from './header.css';

const footer = require('./footer.css');

const Header = () => <_StyledModules styles={[{
  __hash: _globalStyles.__hash,
  __css: _globalStyles.__css
}, {
  __hash: _globalStyles2.__hash,
  __css: _globalStyles2.__css
}, {
  __hash: styles.__hash,
  __css: styles.__css
}, {
  __hash: _styles.__hash,
  __css: _styles.__css
}, {
  __hash: footer.__hash,
  __css: footer.__css
}]}>
  <div className={root} />
</_StyledModules>;

const Footer = () => <_StyledModules styles={[{
  __hash: _globalStyles.__hash,
  __css: _globalStyles.__css
}, {
  __hash: _globalStyles2.__hash,
  __css: _globalStyles2.__css
}, {
  __hash: styles.__hash,
  __css: styles.__css
}, {
  __hash: _styles.__hash,
  __css: _styles.__css
}, {
  __hash: footer.__hash,
  __css: footer.__css
}]}>
  <div className={footer.root} />
</_StyledModules>;

export default class extends React.Component {
  render() {
    return <_StyledModules styles={[{
      __hash: _globalStyles.__hash,
      __css: _globalStyles.__css
    }, {
      __hash: _globalStyles2.__hash,
      __css: _globalStyles2.__css
    }, {
      __hash: styles.__hash,
      __css: styles.__css
    }, {
      __hash: _styles.__hash,
      __css: _styles.__css
    }, {
      __hash: footer.__hash,
      __css: footer.__css
    }]}>
      <div className={styles.helloWorld}>
        <Header />
        Hello World
        {[1, 2, 3].map((value, index) => <_StyledModules key={index} styles={[{
          __hash: _globalStyles.__hash,
          __css: _globalStyles.__css
        }, {
          __hash: _globalStyles2.__hash,
          __css: _globalStyles2.__css
        }, {
          __hash: styles.__hash,
          __css: styles.__css
        }, {
          __hash: _styles.__hash,
          __css: _styles.__css
        }, {
          __hash: footer.__hash,
          __css: footer.__css
        }]}>
          <span>{value}</span>
        </_StyledModules>)}
        <Footer />
      </div>
    </_StyledModules>;
  }
}"
`;
